fvm/src/syscalls/debug.rs

```diff
@@ -21,3 +23,36 @@ pub fn enabled(context: Context<'_, impl Kernel>) -> Result<i32> {
         -1
     })
 }
+
+// TODO: make output path more configurable, maybe add extra gaurds/limitations
+pub fn capture_artifact(
+    context: Context<'_, impl Kernel>,
+    name_off: u32,
+    name_len: u32,
+    data_off: u32,
+    data_len: u32,
+) -> Result<()> {
+    // No-op if disabled.
+    if !context.kernel.debug_enabled() {
+        return Ok(());
+    }
+    let src_dir = std::env::current_dir()
+        .or_fatal()?
+        .canonicalize()
+        .or_fatal()?
+        .join("artifacts");
+    DirBuilder::new()
+        .recursive(true)
+        .create(src_dir.clone())
+        .or_fatal()?;
```

Comment:
This should probably be configured via an environment variable. That way we don't save artifacts when debug is enabled, unless the user explicitly configures where to put them.

Reply:
Should that be a fatal error or fail silently?